bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1982,41 +1982,29 @@ private static boolean checkIsLikeTupleType(Object sourceValue, BTupleType targe
         }
 
         ArrayValue source = (ArrayValue) sourceValue;
-        if (source.size() != targetType.getTupleTypes().size()) {
+        List<BType> targetTypes = new ArrayList<>(targetType.getTupleTypes());
+        int sourceTypeSize = source.size();
+        int targetTypeSize = targetTypes.size();
+        BType targetRestType = targetType.getRestType();
+
+        if (targetRestType == null && sourceTypeSize != targetTypeSize) {
+            return false;
+        }
+        if (targetRestType != null && sourceTypeSize < targetTypeSize) {
             return false;
         }
```

Comment:
Isn't it definitely an error if `sourceTypeSize < targetTypeSize` irrespective of whether or not there is a rest descriptor? Shouldn't we check that and return false as the first check?

Reply:
I guess we can then do 
```suggestion
        if (sourceTypeSize < targetTypeSize) {
            return false;
        }
        
        if (targetRestType == null && sourceTypeSize > targetTypeSize) {
            return false;
        }
```

Reply:
That is correct, changed.